cargo-geiger/src/tree/traversal/dependency_node.rs

```diff
@@ -0,0 +1,246 @@
+        input_edge_direction: EdgeDirection,
+        expected_build_nodes_length: usize,
+        expected_development_nodes_length: usize,
+        expected_normal_nodes_length: usize,
+    ) {
+        let mut inner_graph = petgraph::Graph::<Node, DepKind>::new();
+        let mut nodes = HashMap::<PackageId, NodeIndex>::new();
+
+        let package_ids = create_package_id_vec(7);
+        let print_config = create_print_config(input_edge_direction);
+
+        for package_id in &package_ids {
+            nodes.insert(
+                *package_id,
+                inner_graph.add_node(Node { id: *package_id }),
+            );
+        }
+
+        add_edges_to_graph(
+            &input_directed_edges,
+            &mut inner_graph,
+            &nodes,
+            &package_ids,
+        );
+
+        let graph = Graph {
+            graph: inner_graph,
+            nodes,
+        };
+
+        let dependency_type_nodes_hashmap =
+            construct_dependency_type_nodes_hashmap(
+                &graph,
+                &Node { id: package_ids[0] },
+                &print_config,
+            );
+
+        assert_eq!(
+            dependency_type_nodes_hashmap[&DepKind::Build].len(),
+            expected_build_nodes_length
+        );
+        assert_eq!(
+            dependency_type_nodes_hashmap[&DepKind::Development].len(),
+            expected_development_nodes_length
+        );
+        assert_eq!(
+            dependency_type_nodes_hashmap[&DepKind::Normal].len(),
+            expected_normal_nodes_length
+        );
+    }
+
+    fn add_edges_to_graph(
+        directed_edges: &[(usize, usize, DepKind)],
+        graph: &mut petgraph::Graph<Node, DepKind>,
+        nodes: &HashMap<PackageId, NodeIndex>,
+        package_ids: &[PackageId],
+    ) {
+        for (source_index, target_index, dep_kind) in directed_edges {
+            graph.add_edge(
+                nodes[&package_ids[*source_index]],
+                nodes[&package_ids[*target_index]],
+                *dep_kind,
+            );
+        }
+    }
+
+    fn create_package_id_vec(count: i32) -> Vec<PackageId> {
+        let config = Config::default().unwrap();
+
+        let current_working_dir =
+            env::current_dir().unwrap().join("Cargo.toml");
+
+        let manifest_path_option = Some(current_working_dir);
+
+        let workspace = get_workspace(&config, manifest_path_option).unwrap();
+
+        let package = workspace.current().unwrap();
+
+        let source_id = package.dependencies().first().unwrap().source_id();
+
+        let mut package_id_vec: Vec<PackageId> = vec![];
+
+        for i in 0..count {
+            package_id_vec.push(
+                PackageId::new(
+                    format!("test_name_{}", i),
+                    format!("1.2.{}", i).as_str(),
+                    source_id,
+                )
+                .unwrap(),
+            )
+        }
+
+        package_id_vec
+    }
```

Comment:
Minor, formatting: This function would be easier to read imho without any extra empty lines.

Reply:
Good shout! Have pushed an update.

For the use of `cargo format` would it also be worth drafting a contributing guideline as well?

Reply:
Sure, a short contributing section in the readme file could be useful.